### PR TITLE
Added Neuropod Torchscript export

### PR DESCRIPTION
**Pull request by tgaddair**

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -88,6 +89,12 @@ jobs:
           pip list
         shell: bash
 
+      - name: Install Neuropod backend
+        run: |
+          sudo mkdir -p "$NEUROPOD_BASE_DIR"
+          curl -L https://github.com/uber/neuropod/releases/download/v0.3.0-rc6/libneuropod-cpu-linux-v0.3.0-rc6-torchscript-1.10.2-backend.tar.gz | sudo tar -xz -C "$NEUROPOD_BASE_DIR"
```
Comment (brightsparc) [inline .github/workflows/pytest.yml:95]: Suggest including ENVARS in this curl request  for future upgrades:
* `NEUROPOD_VERISON:  "0.3.0-rc6"`
* `TORCHSCRIPT_VERISON: "1.10.2"`

```
https://github.com/uber/neuropod/releases/download/v${{ env.NEUROPOD_VERISON }}/libneuropod-cpu-linux-v${{ env.NEUROPOD_VERISON }}-torchscript-${{ env.TORCHSCRIPT_VERISON }}-backend.tar.gz
```

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -226,6 +234,12 @@ jobs:
...
```
Comment (brightsparc) [inline .github/workflows/pytest.yml:240]: Same env vars as above.

File: ludwig/utils/neuropod_utils.py [hunk 3]
```diff
@@ -1,208 +1,119 @@
...
+        super().__init__()
+        self.inference_module = inference_module
+
+    def forward(self, {input_signature}):
+        inputs = {input_dict}
+        results = self.inference_module(inputs)
+        #for name, preds in results.items():
+        #    for k, v in preds.items():
+        #        if v.dtype == torch.bool:
+        #            preds[k] = v.to(dtype=torch.int32)
```
Comment (brightsparc) [inline ludwig/utils/neuropod_utils.py:30]: Is this comment here for a reason?

File: ludwig/utils/neuropod_utils.py [hunk 4]
```diff
@@ -1,208 +1,119 @@
...
+        return {output_dicts}
+"""
+
+
+def _get_input_signature(config: Dict[str, Any]) -> str:
+    args = []
+    for feature in config["input_features"]:
+        name = feature[NAME]
+        args.append(f"{name}: Union[List[str], torch.Tensor]")
```
Comment (brightsparc) [inline ludwig/utils/neuropod_utils.py:39]: Optionally remove Union[] with explicit typing:

```
        name = feature[NAME]
        dt = 'List[str]' if feature[TYPE] in ['category'] else 'torch.Tensor'
        args.append(f"{name}: {dt}")
```

**Review by brightsparc — APPROVED** [review]
Suggested some minor improvements for env vars.

File: ludwig/utils/neuropod_utils.py (hunk 3 above)
Comment (tgaddair) [reply ludwig/utils/neuropod_utils.py:30]: Experimental code for binary types. Will remove for now.

File: ludwig/utils/neuropod_utils.py (hunk 4 above)
Comment (tgaddair) [reply ludwig/utils/neuropod_utils.py:39]: Some of these can be both (like bools), category should also support a tensor type (as ints) in the future. So will leave as is for now, but we can revisit.

**Pull request merged**